lib/Exporter/ONNXModelWriter.cpp

```diff
@@ -0,0 +1,1367 @@
+    const Node *node, GraphType &graph, ReporterType &) {
+  auto *proto = graph.add_node();
+  // Find dictionary entires
+  const auto *CQC = llvm::dyn_cast<ChannelwiseQuantizedConvolutionNode>(node);
+  RETURN_ERR_IF_NOT(
+      CQC, "Invalid node type, expected ChannelwiseQuantizedConvolutionNode.");
+
+  auto *attr = proto->add_attribute();
+  attr->set_name("kernel_shape");
+  attr->set_type(AttrType::INTS);
+  for (auto value : CQC->getKernels()) {
+    attr->add_ints(value);
+  }
+
+  attr = proto->add_attribute();
+  attr->set_name("strides");
+  attr->set_type(AttrType::INTS);
+  for (auto value : CQC->getStrides()) {
+    attr->add_ints(value);
+  }
+
+  attr = proto->add_attribute();
+  attr->set_name("pads");
+  attr->set_type(AttrType::INTS);
+  for (auto value : CQC->getPads()) {
+    attr->add_ints(value);
+  }
+
+  attr = proto->add_attribute();
+  attr->set_name("group");
+  attr->set_type(AttrType::INT);
+  attr->set_i(CQC->getGroup());
+
+  attr = proto->add_attribute();
+  attr->set_name("group_wise");
+  attr->set_type(AttrType::INT);
+  attr->set_i(CQC->getGroupwise() ? 1 : 0);
+
+  return writeAllWithNode(node->getName(), node, proto);
+}
+
+llvm::Error ONNXModelWriter::writeSplat(const Node *node, GraphType &graph,
+                                        ReporterType &) {
+  auto *proto = graph.add_node();
+  // Find dictionary entires
+  const auto *SN = llvm::dyn_cast<SplatNode>(node);
+  RETURN_ERR_IF_NOT(SN, "Invalid node type, expected SplatNode.");
+
+  auto *attr = proto->add_attribute();
+  attr = proto->add_attribute();
+  attr->set_name("value");
+  attr->set_type(AttrType::FLOAT);
+  attr->set_f(SN->getValue());
+
+  return writeAllWithNode(node->getName(), node, proto);
+}
+
+llvm::Error ONNXModelWriter::writeCPUMaxSplat(const Node *node,
+                                              GraphType &graph,
+                                              ReporterType &) {
+  auto *proto = graph.add_node();
+  // Find dictionary entires
+  const auto *SN = llvm::dyn_cast<CPUMaxSplatNode>(node);
+  RETURN_ERR_IF_NOT(SN, "Invalid node type, expected CPUMaxSplatNode.");
+
+  auto *attr = proto->add_attribute();
+  attr = proto->add_attribute();
+  attr->set_name("value");
+  attr->set_type(AttrType::FLOAT);
+  attr->set_f(SN->getSplatValue());
+
+  return writeAllWithNode(node->getName(), node, proto);
+}
+
+llvm::Error ONNXModelWriter::writeCPUConvDKKC8(const Node *node,
+                                               GraphType &graph,
+                                               ReporterType &) {
+  auto *proto = graph.add_node();
+  //Find dictionary entires
+  const auto *CDN = llvm::dyn_cast<CPUConvDKKC8Node>(node);
+  RETURN_ERR_IF_NOT(CDN, "Invalid node type, expected CPUConvDKKC8Node.");
+
+  auto *attr = proto->add_attribute();
+  attr->set_name("kernel_shape");
+  attr->set_type(AttrType::INTS);
+  for (auto value : CDN->getKernels()) {
+    attr->add_ints(value);
+  }
+
+  attr = proto->add_attribute();
```

Comment:
Seems like this whole block is often repeated. Seems like a good candidate for introducing a helper method. There are many similar, but slightly different blocks that could also benefit from helpers.